### DISCUSSION TITLE
How to get Element dimensions?

### DISCUSSION THREAD

**Discussion by kajmaj87** [ArthurSonzogni/FTXUI#395] [Q&A] [answered]
I'm building a simple interface with 4 sections of text (top, bottom, left, right) and a canvas in the middle of them all. All of the sections are resizable. 
I would like to know at runtime how big my canvas should be and how to interpret my mouse events - would have to map my mouse to position on canvas. Is there a way to determine the needed size and mouse offsets (basically top/bottom/left/right sizes)?

**Comment by ArthurSonzogni** [comment, marked as answer]
There is a 'canvas' function taking a callback. Instead of providing your own canvas of fixed size, one filling the full area is provided to you:
https://arthursonzogni.github.io/FTXUI/namespaceftxui.html#a8bef154e558868d259c52380e1e5ead9

Alternatively, you can wrap your element with 'reflect'. It will write the size/position of the box where it has been rendered to:
https://arthursonzogni.github.io/FTXUI/namespaceftxui.html#aaff8245861617a3d9e846e99de582a63
